corehq/util/context_managers.py

```diff
@@ -40,3 +45,14 @@ def catch_signal(signal):
     signal.connect(handler)
     yield handler
     signal.disconnect(handler)
+
+
+@contextmanager
+def prevent_parallel_execution(cache_key, timeout=DEFAULT_PARALLEL_EXECUTION_TIMEOUT):
+    if cache.get(cache_key, False):
```

Comment:
any reason not to use `CriticalSection`:
```python
with CriticalSection('my key', fail_hard=True, timeout=60 * 60, block=False):
    pass
```

Reply:
Thanks for sharing that @snopoke , i was not aware we already have something built for that kind of work
I see its doing a little more than what we need like datadog reporting. I was trying to build on lines like `pending_location_import_download_id`.
But if you think we should be now follow `CriticalSection` as the way to do, i am happy to update this to use it, although one thing it would do is things getting reported to DD that wont be used unless we edit it to skip DD reporting.

Reply:
just thinking about using existing components rather than building new ones

Reply:
Agreed with that. I just thought `CriticalSection` comes in with more than we need here.